app/session.server.ts

```diff
@@ -27,13 +36,48 @@ const { getSession, commitSession, destroySession } =
 
 export { getSession, commitSession, destroySession };
 
+/**
+ * Returns information from GitHub about the user associated with the access
+ * token. If we're running this code in development using the Firebase
+ * emulators, we will fake the data.
+ */
+async function getGitHubUserDataFromClaims(
+  accessToken: string,
+  decodedClaims: DecodedIdToken
+): Promise<{ login: string; picture?: string; email?: string }> {
+  if (
+    process.env.NODE_ENV === "development" &&
+    accessToken === FIREBASE_FAKE_ACCESS_TOKEN
+  ) {
+    return {
+      login: "test-" + decodedClaims.name.split(" ").join("-").toLowerCase(),
+      picture: decodedClaims.picture,
+      email: decodedClaims.email,
+    };
+  } else {
+    // We use the accessToken to fetch the user from GitHub. This is required
+    // because it's the only way to get the "login" field! However, we could
+    // probably save it once we've fetched it once.
+    const octokit = new Octokit({
+      auth: accessToken,
+    });
+    const { data } = await octokit.request("GET /user");
+    return {
+      login: data.login,
+      picture: data.avatar_url,
+      email: data.email || undefined,
```

Comment:
I wonder if it would make more sense to set email to `null` here? We aren't saying that it's undefined, right? We're really saying that we know for a fact there is no email.

Reply:
You're right, I'll change that.